src/fw2tar

```diff
@@ -178,43 +184,61 @@ def _tar_fs(rootfs_dir, tarbase):
         print(f"Error archiving root filesystem {rootfs_dir}: {tar_result.stderr}")
         return
 
-    # Now, compress the tar archive using gzip with the --no-name option
+    # Now, compress the tar archive using gzip
     # Output filename will be tarbase.tar.gz
-    gzip_command = ["gzip", "--no-name", "-f", uncompressed_outfile]
+    with open(uncompressed_outfile, 'rb') as f_in, gzip.open(f"{uncompressed_outfile}.gz", 'wb') as f_out:
+        shutil.copyfileobj(f_in, f_out)
 
-    # Execute the gzip command
-    gzip_result = subprocess.run(gzip_command, capture_output=True, text=True)
-    if gzip_result.returncode != 0:
-        print(f"Error compressing tar archive {uncompressed_outfile}: {gzip_result.stderr}")
-        return
+    # Remove tarbase.tar in favor of just keeping the gzipped version
+    os.remove(uncompressed_outfile)
 
     # Chmod the tar.gz file to 644
     os.chmod(f"{uncompressed_outfile}.gz", 0o644)
 
 def _extract(extractor, infile, extract_dir, log_file):
     try:
         if extractor == "unblob":
-            subprocess.run(["unblob",
-                            "--log", log_file,
-                            "--extract-dir", str(extract_dir),
-                            infile], check=True,
-                            capture_output=True, text=True)
+            # XXX: Unblob initializes multiprocessing during import so we need to
+            #      monkey-patch out re-initializing mp before we can use unblob
+            def set_start_method_nop(method, force=False):
+                pass
+            multiprocessing.set_start_method = set_start_method_nop
+
+            from unblob.processing import ExtractionConfig, process_file
+            from unblob.logging import configure_logger
+
+            configure_logger(0, extract_dir, Path(log_file))
+            config = ExtractionConfig(extract_root=extract_dir, entropy_depth=1)
+            process_file(config, Path(infile))
         elif extractor == "binwalk":
-            subprocess.run(["binwalk", "--run-as=root",
-                            "--preserve-symlinks",
-                            "-eM",
-                            "--log", log_file,
-                            "-q", infile,
-                            "-C", str(extract_dir)],
-                            check=True, capture_output=True,
-                            text=True)
+            import binwalk
+
+            modules = binwalk.scan(
```

Comment:
Can we get more involved in the recursive part of the extraction when using the APIs? In particular, I'm interested in the case where both extractors identify the same blob that they then recurse into. I want us to be able to extract every unique blob with both extractors instead of independently doing the recursive extractions